### PR TITLE
Profile frontend

### PR DESCRIPTION
**Pull request by rain72510**
Change layout of sidebar and readme

### CODE REVIEW

File: frontend/src/pages/CreateContestPage.tsx [hunk 1]
```diff
@@ -65,8 +67,9 @@ const CreateContestPage = () => {
 								console.log("body: ", body)
```
Comment (lolainta) [inline frontend/src/pages/CreateContestPage.tsx:73]: Remove debug msgs

File: frontend/src/pages/CreateContestPage.tsx [hunk 2]
```diff
@@ -91,89 +95,111 @@ const CreateContestPage = () => {
...
+														value={values.name}
+														name="name"
+														onChange={handleChange}
+														onBlur={handleBlur}
+													/>
 
-													{touched.date && errors.date && (
+													{touched.name && errors.name && (
 														<FormHelperText error>
 															{' '}
```
Comment (lolainta) [inline frontend/src/pages/CreateContestPage.tsx:113]: Clean it up

File: frontend/src/pages/CreateContestPage.tsx [hunk 3]
```diff
@@ -91,89 +95,111 @@ const CreateContestPage = () => {
...
+														name="overview"
 														onChange={handleChange}
 														onBlur={handleBlur}
+														multiline
 													/>
 
-													{touched.name && errors.name && (
+													{touched.overview && errors.overview && (
 														<FormHelperText error>
 															{' '}
```
Comment (lolainta) [inline frontend/src/pages/CreateContestPage.tsx:133]: dito

File: frontend/src/pages/CreateContestPage.tsx [hunk 4]
```diff
@@ -91,89 +95,111 @@ const CreateContestPage = () => {
...
+															value={values.date}
+															// onChange={(e) => setFieldValue("date", parse(e), true)}
+															onChange={(e) => e}
+															// renderInput={(params) => <TextField {...params} />}
+															slotProps={{textField: {required: true}}}
+														/>
+													</LocalizationProvider>
+
+													{touched.date && errors.date && (
+														<FormHelperText error>
```
Comment (lolainta) [inline frontend/src/pages/CreateContestPage.tsx:153]: dito

**Review by rain72510 — COMMENTED** [review]
Also can use session log in.

File: .gitignore
Comment (lolainta) [inline .gitignore:1]: don't change unrelated files, it might cause conflict.

File: frontend/src/App.tsx
Comment (lolainta) [inline frontend/src/App.tsx:1]: remove debug code and refactor all `fetch`s to another file

Comment (lolainta) [reply frontend/src/App.tsx:1]: Do you want to take axios a try?
https://axios-http.com/docs/intro

File: frontend/src/components/Header.tsx [hunk 5]
```diff
@@ -114,18 +116,19 @@ const Header: FC<HeaderProps> = ({ setSideBarOpen, setAuthorized }) => {
                                 onClick={() => {
                                   handleClose();
                                   setAuthorized(false);
-                                  fetch("http://localhost:8080/api/logout", {
+                                  fetch(`http://localhost:8080/${api.user.logout}`, {
```
Comment (lolainta) [inline frontend/src/components/Header.tsx:119]: Use a config.tsx for backend host url, and import this configuration file in `api.tsx` only

File: frontend/src/components/Sidebar.tsx [hunk 6]
```diff
@@ -49,41 +122,20 @@ const Sidebar: FC<SidebarProps> = ({ sideBarOpen, setSideBarOpen }) => {
...
 	return (
-
 		<ClickAwayListener onClickAway={handleClose}>
-		<Box sx={{position:"fixed",	zIndex: 10}}>
-			{/* {sideBarOpen && list()} */}
-			{list()}
-		</Box>
-
+			<Box sx={{position:"fixed",	zIndex: 10}}>
+				{/* {sideBarOpen && list()} */}
```
Comment (lolainta) [inline frontend/src/components/Sidebar.tsx:136]: remove unused codes

File: frontend/src/pages/ChangeInfo.tsx [hunk 7]
```diff
@@ -30,47 +32,79 @@ const ChangeInfo = () => {
...
+							method: "GET",
+							credentials: "include",
+						})
+						.then((res) => res.json())
+						.then((resjson) => {
+							if (resjson["result"] && resjson["result"] == "success") {
+								body.append("username", resjson["name"]);
+							}
+						});
+						await fetch(`http://localhost:8080/${api.user.modifyInfo}`, {
```
Comment (lolainta) [inline frontend/src/pages/ChangeInfo.tsx:91]: dito

File: frontend/src/pages/ChangeInfo.tsx [hunk 8]
```diff
@@ -30,47 +32,79 @@ const ChangeInfo = () => {
...
-							console.log(err);
-						}
+						
+						// body.append("nickname", values.nickname);
+						body.append("overview", values.overview);
+						body.append("oriPassword", values.oriPassword);
+						body.append("organization", values.organization);
+						console.log("body: ", body);
+						var id;
+						await fetch(`http://localhost:8080/${api.user.getUserID}`, {
```
Comment (lolainta) [inline frontend/src/pages/ChangeInfo.tsx:71]: config file with url

File: frontend/src/pages/ChangeInfo.tsx [hunk 9]
```diff
@@ -30,47 +32,79 @@ const ChangeInfo = () => {
...
+							method: "GET",
+							credentials: "include",
+						})
+						.then((res) => res.json())
+						.then((resjson) => {
+							if (resjson["uid"]) {
+								id = resjson["uid"];
+							}
+						});
+						await fetch(`http://localhost:8080/${api.user.info}/${id}`, {
```
Comment (lolainta) [inline frontend/src/pages/ChangeInfo.tsx:81]: dito

File: frontend/src/pages/ContestPage.tsx [hunk 10]
```diff
@@ -56,7 +58,8 @@ import { useNavigate } from 'react-router-dom';
 			"date": "2017-04-23 10:30",
```
Comment (lolainta) [inline frontend/src/pages/ContestPage.tsx:58]: move to data.js for testing

File: frontend/src/pages/ContestPage.tsx [hunk 11]
```diff
@@ -263,14 +278,51 @@ const ContestPage = () => {
...
+							<Grid container justifyContent="center">
+								<Grid item sx={{width: "70%"}}>
+									<Button 
+										variant="text"
+										sx={{ color: "#2074d4", width: "100%" }}
+										onClick={() => {
+											const body = new FormData();
+											body.append("competitionID", "aaaa");
+
+											fetch(`http://localhost:8080/${api.competition.join}`, {
```
Comment (lolainta) [inline frontend/src/pages/ContestPage.tsx:291]: config

File: frontend/src/pages/ContestPage.tsx [hunk 12]
```diff
@@ -263,14 +278,51 @@ const ContestPage = () => {
...
+												method: "POST",
+												credentials: "include",
+												body: body,
+											})
+											.then((res) => {
+												return res.json();
+											})
+											.then((resjson) => {
+		
+												console.log(resjson);
```
Comment (lolainta) [inline frontend/src/pages/ContestPage.tsx:301]: remove debug msgs

File: frontend/src/pages/CreateContestPage.tsx [hunk 13]
```diff
@@ -57,123 +64,145 @@ const CreateContestPage = () => {
...
-										if (resjson["result"] && resjson["result"] === "success") {
-											console.log("Log In Success");
-										} else {
-											console.log("Too bad QQ");
-										}
-									});
-								} catch (err) {
-									console.log(err);
-								}
+								fetch(`http://localhost:8080/${api.competition.create}`, {
```
Comment (lolainta) [inline frontend/src/pages/CreateContestPage.tsx:75]: config

File: frontend/src/pages/LoginPage.tsx [hunk 14]
```diff
@@ -44,6 +45,22 @@ const LoginPage: FC<LoginPageProps> = ({setAuthorized, setPath}) => {
 
 	const navigate = useNavigate();
 
+	fetch(`http://localhost:8080/${api.user.getUserID}`, {
```
Comment (lolainta) [inline frontend/src/pages/LoginPage.tsx:48]: config

File: frontend/src/pages/SignupPage.tsx [hunk 15]
```diff
@@ -22,7 +22,8 @@ import * as Yup from 'yup';
 import { Formik } from 'formik';
 import { FormControl } from '@mui/material';
 
-import * as AES from 'crypto-js/aes';
+import api from '../util/api';
+import routing from '../util/routing';
 
 interface SignupPageProps {
 	setPath: Dispatch<SetStateAction<string>>;
```
Comment (lolainta) [inline frontend/src/pages/SignupPage.tsx:29]: space or tab?

**Comment by lolainta** [comment]
This is a roughly review only, please resolve similar issues and I'll look ahead.

**Comment by lolainta** [comment]
Reopen the issue when you done it. Look #13 comments for detail actions
